test/built-ins/RegExp/prototype/dotAll/this-val-invalid-obj.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Invoked on an object without an [[OriginalFlags]] internal slot
+esid: pending
+info: >
+    get RegExp.prototype.dotAll
+
+    1. Let R be the this value.
+    2. If Type(R) is not Object, throw a TypeError exception.
+    3. If R does not have an [[OriginalFlags]] internal slot, throw a TypeError
+       exception.
+features: [regexp-dotall]
+---*/
+
+var dotAll = Object.getOwnPropertyDescriptor(RegExp.prototype, 'dotAll').get;
+
+assert.throws(TypeError, function() {
+  dotAll.call({});
+}, 'ordinary object');
+
+assert.throws(TypeError, function() {
+  dotAll.call([]);
+}, 'array exotic object');
+
+assert.throws(TypeError, function() {
+  dotAll.call(arguments);
```

Comment:
why arguments? This is already an Object. Maybe it would be better replacing it with `RegExp` itself or even better: [`RegExp.prototype`](https://tc39.github.io/ecma262/#sec-properties-of-the-regexp-prototype-object) that explicitly doesn't have any RegExp internals.